drivers/misc/ipts/companion/ipts-surface.c

```diff
@@ -1,44 +1,69 @@
 #include <linux/acpi.h>
 #include <linux/firmware.h>
-#include <linux/intel_ipts_fw.h>
-#include <linux/intel_ipts_if.h>
+#include <linux/ipts.h>
+#include <linux/ipts-companion.h>
 #include <linux/module.h>
 #include <linux/platform_device.h>
 
 #define IPTS_SURFACE_FW_PATH_FMT "intel/ipts/%s/%s"
 
-#define __IPTS_SURFACE_FIRMWARE(X, Y)				\
-	MODULE_FIRMWARE("intel/ipts/" X "/" Y)
-
-#define IPTS_SURFACE_FIRMWARE(X)				\
-	__IPTS_SURFACE_FIRMWARE(X, "config.bin");		\
-	__IPTS_SURFACE_FIRMWARE(X, "intel_desc.bin");		\
-	__IPTS_SURFACE_FIRMWARE(X, "intel_fw_config.bin");	\
-	__IPTS_SURFACE_FIRMWARE(X, "vendor_desc.bin");		\
-	__IPTS_SURFACE_FIRMWARE(X, "vendor_kernel.bin")
-
-IPTS_SURFACE_FIRMWARE("MSHW0076");
-IPTS_SURFACE_FIRMWARE("MSHW0078");
-IPTS_SURFACE_FIRMWARE("MSHW0079");
-IPTS_SURFACE_FIRMWARE("MSHW0101");
-IPTS_SURFACE_FIRMWARE("MSHW0102");
-IPTS_SURFACE_FIRMWARE("MSHW0103");
-IPTS_SURFACE_FIRMWARE("MSHW0137");
+#define IPTS_SURFACE_FIRMWARE(X)					\
+	MODULE_FIRMWARE("intel/ipts/" X "/config.bin");			\
+	MODULE_FIRMWARE("intel/ipts/" X "/intel_desc.bin");		\
+	MODULE_FIRMWARE("intel/ipts/" X "/vendor_desc.bin");		\
+	MODULE_FIRMWARE("intel/ipts/" X "/vendor_kernel.bin");		\
 
 int ipts_surface_request_firmware(const struct firmware **fw, const char *name,
-	struct device *device, void *data)
+		struct device *device, ipts_companion_t *companion)
 {
 	char fw_path[MAX_IOCL_FILE_PATH_LEN];
 
-	if (data == NULL) {
+	if (companion == NULL || companion->data == NULL) {
 		return -ENOENT;
 	}
 
 	snprintf(fw_path, MAX_IOCL_FILE_PATH_LEN, IPTS_SURFACE_FW_PATH_FMT,
-		(const char *)data, name);
+			(const char *)companion->data, name);
 	return request_firmware(fw, fw_path, device);
 }
 
+static ipts_bin_fw_info_t ipts_surface_vendor_kernel = {
+	.fw_name = "vendor_kernel.bin",
+	.vendor_output = -1,
+	.num_of_data_files = 3,
+	.data_file = {
+		{
+			.io_buffer_type = IPTS_CONFIGURATION,
+			.flags = IPTS_DATA_FILE_FLAG_NONE,
+			.file_name = "config.bin",
+		},
+
+		// The following files are part of the config, but they don't
+		// exist, and the driver never requests them.
```

Comment:
If this is true, can we comment these both out and prove that?

Reply:
I would rather keep them, as a reference, so that we don't have to reverse engineer the binary file again at some point. That they aren't requested can already be proven, because the files don't exist, and there are no errors about them missing in the log.

Reply:
I think we should keep them for now. We can always remove them in a future commit, but it's probably a good idea to have them in the history.